corehq/apps/cloudcare/views.py

```diff
@@ -587,36 +596,26 @@ def _fail(error):
     if not toggles.SESSION_ENDPOINTS.enabled_for_request(request):
         return _fail(_("Linking directly into Web Apps has been disabled."))
 
-    build = _fetch_build(domain, request.couch_user.username, app_id)
-    if not build:
+    build_id = _fetch_build_id(domain, request.couch_user.username, app_id)
+    if not build_id:
         # These links can be used for cross-domain web apps workflows, where a link jumps to the
         # same screen but in another domain's corresponding app. This works if both the source and
         # target apps are downstream apps that share an upstream app - the link references the upstream app.
         from corehq.apps.linked_domain.applications import get_downstream_app_id_map
         id_map = get_downstream_app_id_map(domain)
         if app_id in id_map:
             if len(id_map[app_id]) == 1:
-                build = _fetch_build(domain, request.couch_user.username, id_map[app_id][0])
-        if not build:
+                build_id = _fetch_build_id(domain, request.couch_user.username, id_map[app_id][0])
+        if not build_id:
             return _fail(_("Could not find application."))
 
-    valid_endpoint_ids = set()
```

Comment:
Formplayer already checks that the endpoint is valid: https://github.com/dimagi/formplayer/blob/0ddd54786757e5c16980791b98c88ebcee03efcd/src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java#L678-L681